pandas/core/groupby/groupby.py

```diff
@@ -318,14 +318,13 @@ class providing the base-class of operations.
 """
 
 _transform_template = """
-Call function producing a like-indexed %(klass)s on each group and
-return a %(klass)s having the same indexes as the original object
-filled with the transformed values
+Apply function ``func`` column-by-column to the GroupBy object and return a %(klass)s
+with the same number of indices as the group.
```

Comment:
The part of the phrase is referring to the entire return of the transform method. Thus, saying "as the group" here doesn't make sense - there can be multiple groups, so there is no "_the_ group". Also, @jreback was commenting that the return must have the same index (not just the same number of elements!) as the input.